### PR TITLE
Added New Theme - ExtraDark

### PR DESCRIPTION
**Pull request by SobhanBera**
### Updates:
This is a new theme I have tried to add. I am continuously working on this theme to give the same primary colour to the borders, shadows, header elements, album name colour, title colours, etc. Most importantly the player component to change the primary colour.

Actually, I need some source to change the primary colour of the player. Currently, it could be changed using index.less which is in `node_modules` or by changing the primary-colour property at `webpack.config.js`.

### Screenshots:

ArcoLinux-2021-03-24-1616609456_screenshot_1920x1080

ArcoLinux-2021-03-24-1616609479_screenshot_1920x1080

### CODE REVIEW

File: ui/src/layout/PersonalMenu.js [hunk 1]
```diff
@@ -3,11 +3,14 @@ import { MenuItemLink, useTranslate } from 'react-admin'
 import { makeStyles } from '@material-ui/core'
 import TuneIcon from '@material-ui/icons/Tune'
 
-const useStyles = makeStyles((theme) => ({
-  menuItem: {
-    color: theme.palette.text.secondary,
-  },
-}))
+const useStyles = makeStyles((theme) => {
+  console.log('THEMEMEMEMEMEMEMMEMEMEMEME', theme.palette)
```
Comment (deluan) [inline ui/src/layout/PersonalMenu.js:7]: Please remove.

File: ui/src/themes/index.js [hunk 2]
```diff
@@ -1,5 +1,35 @@
...
+ * working -
+ * if(value[0][0] === value[0][1])
+ *    return value[0][2]
+ * else if(value[1][0] === value[1][1])
+ *    return value[1][2]
+ * else
+ *    return defaultValue
+ */
+export function themeColorSelector(defaultValue, ...values) {
+  console.log('DEF', defaultValue)
```
Comment (deluan) [inline ui/src/themes/index.js:27]: Please remove.

File: ui/src/themes/index.js [hunk 3]
```diff
@@ -1,5 +1,35 @@
...
+  for (let val in values) {
+    if (values[val][0] === values[val][1]) {
+      console.log('VALUE', values[val][0], values[val][1], values[val][2])
```
Comment (deluan) [inline ui/src/themes/index.js:30]: Please remove.

File: ui/src/album/AlbumGridView.js [hunk 4]
```diff
@@ -53,7 +54,15 @@ const useStyles = makeStyles((theme) => ({
...
     fontSize: '12px',
-    color: theme.palette.type === 'dark' ? '#c5c5c5' : '#696969',
+    color: themeColorSelector(
+      '#696969',
+      [
+        theme.palette?.extraAttribute.theme,
+        'extradark',
+        theme.palette?.extraAttribute.subtitle,
+      ],
+      [theme.palette.type, 'dark', '#c5c5c5']
```
Comment (deluan) [inline ui/src/album/AlbumGridView.js:64]: Using code to select a colour is really bad (it was already bad before :) ). I think this should become a field in the `theme`, or maybe try to find one colour in the palette that is similar to this one (in the existing themes), and override in your new theme.

File: ui/src/album/AlbumGridView.js [hunk 5]
```diff
@@ -53,7 +54,15 @@ const useStyles = makeStyles((theme) => ({
   },
   albumSubtitle: {
     fontSize: '12px',
-    color: theme.palette.type === 'dark' ? '#c5c5c5' : '#696969',
+    color: themeColorSelector(
```
Comment (deluan) [inline ui/src/album/AlbumGridView.js:57]: I don't get what this is doing... Not straightforward. Take a look at the comment below, maybe just retrieveing this colour from the theme is a better choice.

File: ui/src/themes/extradark.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      contrastText: '#909090',
+    },
+    secondary: blue,
+    extra: {
+      bordercolors: '#202020',
+    },
+    type: 'dark',
+    extraAttribute: {
+      theme: 'extradark',
+      lines: '#353535',
```
Comment (deluan) [inline ui/src/themes/extradark.js:21]: Looks like this attribute is not used.

File: ui/src/themes/extradark.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      paper: '#000000',
+      default: '#000000',
+    },
+    primary: {
+      main: '#0f60b6',
+      contrastText: '#909090',
+    },
+    secondary: blue,
+    extra: {
+      bordercolors: '#202020',
```
Comment (deluan) [inline ui/src/themes/extradark.js:16]: Looks like this attribute is not used (is it?)

File: ui/src/themes/index.js [hunk 8]
```diff
@@ -1,5 +1,35 @@
...
+/**
+ * @param {*} theme_colors
+ * @param  {...array} this is rest of value which will be returned only if condition are true.
+ * let the arguments provided as -
+ * anyDomEle: {
+ *  color: themeColorSelector('#696969',
+ *                              [theme.palette.extraAttribute.theme, 'extradark', theme.palette.extraAttribute.subtitle],
+ *                              [theme.palette.type, 'dark', '#c5c5c5'])
+ * }
+ * looks complicated but working is simple the function accept a default value, then array of array of 3 elements priority vise, this array elements are as follows [variable, is_variable_equals_to, this_variable]
```
Comment (deluan) [inline ui/src/themes/index.js:17]: IMHO, It **is** complicated.... :)

**Review by deluan — CHANGES_REQUESTED** [review]
I know @jvoisin don't like dark themes, but I liked this. Maybe call it "Night" or "Black"?

What I don't like is having logic to select colours. Don't get me wrong: It was already bad before (mea culpa). But I don't want to make it even worse. What happens if we add a new theme? More  code changes? We need to abstract away all colours, sizes, fonts, paddings, etc... into the theme JSON.

By the way, how did you changed the player's colour? Changing files under `node_modules` is not a solution....

File: ui/src/layout/PersonalMenu.js (hunk 1 above)
Comment (SobhanBera) [reply ui/src/layout/PersonalMenu.js:7]: OK I will remove this

File: ui/src/themes/index.js (hunk 2 above)
Comment (SobhanBera) [reply ui/src/themes/index.js:27]: I was just debugging. I will remove it and make a PR.

File: ui/src/themes/index.js (hunk 3 above)
Comment (SobhanBera) [reply ui/src/themes/index.js:30]: Yep!

File: ui/src/album/AlbumGridView.js (hunk 4 above)
Comment (SobhanBera) [reply ui/src/album/AlbumGridView.js:64]: OK, that seems a great way to select colours. Thanks for the suggestion I will do it.

**Comment by SobhanBera** [comment]
I will resolve all the changes and make a PR soon. Sorry for such silly and nonsense mistake I have done. That's my mistake and I will resolve them all.

**Comment by SobhanBera** [comment]
> I know @jvoisin don't like dark themes, but I liked this. Maybe call it "Night" or "Black"?
> 
> What I don't like is having logic to select colours. Don't get me wrong: It was already bad before (mea culpa). But I don't want to make it even worse. What happens if we add a new theme? More code changes? We need to abstract away all colours, sizes, fonts, paddings, etc... into the theme JSON.
> 
> By the way, how did you changed the player's colour? Changing files under `node_modules` is not a solution....

No, I haven't touched the node_modules folder. I know that makes no sense to edit node_modules to edit the main code. I just changed the colour in the Player component. And again sorry for the mistakes, I will resolve them all and get back to you.

File: ui/src/themes/extradark.js (hunk 6 above)
Comment (SobhanBera) [reply ui/src/themes/extradark.js:21]: yeah I have just tried to change the border and lines under text color.

**Pull request closed without merging**